lib/utils.sh

```diff
@@ -2,12 +2,16 @@
 
 get_ip()
 {
-    IP=$(sudo ifconfig | fgrep "inet " | egrep -v "inet (addr:)?(127|192)\." | egrep -o "inet (addr:)?([0-9]+\.?){4}" | egrep -o "([0-9]+\.?){4}" | head -n 1)
-
-    if [ ! -z "$IP" ]; then
-        echo "$IP"
+    if AWS_IP=$(curl -f -s --connect-timeout 5 http://169.254.169.254/latest/meta-data/public-ipv4); then
```

Comment:
This URL doesn't seem to be loading for me, is it an Amazon-specific URL?

Reply:
@alexbarnsley yes this is specific to AWS.

Reply:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html

Reply:
Thanks for clarifying. It may be worth a comment, in case someone else tries the same thing